core/src/test/scala/akka/persistence/cassandra/query/EventsByPersistenceIdSpec.scala

```diff
@@ -313,6 +360,36 @@ class EventsByPersistenceIdSpec
       probe
         .request(10)
         .expectNextN((22 to 31).map(i => s"q-$i"))
+
+      probe.cancel()
+    }
+
+    "find all events when PersistAll spans partition boundaries" in {
+      val ref = setup("q2", 10)
+      // partition 0
+      ref ! TestActor.PersistAll((11 to 15).map(i => s"q2-$i"))
+      expectMsg("PersistAll-done")
+      // partition 1
+      ref ! TestActor.PersistAll((16 to 17).map(i => s"q2-$i"))
+      expectMsg("PersistAll-done")
+      // all these go into partition 2
+      ref ! TestActor.PersistAll((18 to 31).map(i => s"q2-$i"))
+      expectMsg("PersistAll-done")
+      ref ! "q2-32"
+      expectMsg("q2-32-done")
+
+      val src = queries.eventsByPersistenceId("q2", 0L, Long.MaxValue)
+      val probe = src.map(_.event).runWith(TestSink.probe[Any])
+        .request(10)
+        .expectNextN((1 to 10).map(i => s"q2-$i"))
+        .expectNoMsg(100.millis)
+        .request(10)
+        .expectNextN((11 to 20).map(i => s"q2-$i"))
+        .request(100)
+        .expectNextN((21 to 32).map(i => s"q2-$i"))
+
```

Comment:
forgot, this test should be expanded to write more events here (find new events)